src/wordlift/mappings/class-mappings-rest-controller.php

```diff
@@ -142,8 +142,70 @@ public static function register_route_callback() {
 				},
 			)
 		);
+
+        // Register rest endpoint to get the terms.
+        register_rest_route(
+            WL_REST_ROUTE_DEFAULT_NAMESPACE,
+            'mappings/get_taxonomy_terms',
+            array(
+                'methods'             => WP_REST_Server::CREATABLE,
+                'callback'            => 'Wordlift\Mappings\Mappings_REST_Controller::get_taxonomy_terms_for_the_posted_taxonomy',
+                'permission_callback' => function () {
+                    return current_user_can( 'manage_options' );
+                },
+            )
+        );
 	}
 
+    /**
+     * Get the taxonomy & terms
+     *
+     * @param WP_REST_Request $request {@link WP_REST_Request instance}.
+     *
+     * @return array The array of the taxonomies & terms.
+     */
+    public static function get_taxonomy_terms_for_the_posted_taxonomy( $request ) {
+        $taxonomy_terms = array();
+        $post_taxonomies = get_taxonomies( array(), 'objects' );
+
+        foreach ( $post_taxonomies as $post_taxonomy ) {
```

Comment:
Dont add <optgroup> if there is no <option> present inside that group, because the group cant be selected by user, only entries can be selected.